src/plugins/saved_objects/server/routes/find.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { SavedObjectsRouter } from '../types';
+import type { SavedObjectCommon, FindResponseHTTP } from '../../common';
+
+export const registerFindRoute = (router: SavedObjectsRouter) => {
+  router.get(
+    {
+      path: '/api/saved-objects/find',
```

Comment:
1. use `/internal` instead of `/api` (to avoid regretting it later when we'll have to maintain it publicly)
2. at some point the `savedObjects` plugin will go away. I would future-proof by adding a dedicated path prefix: `/saved-objects-finder` instead of `/saved-objects`, just to be sure.

1+2 => `/internal/saved-objects-finder/find`